plugin/inlay_hint.py

```diff
@@ -112,40 +110,51 @@ def format_inlay_hint_tooltip(tooltip: Optional[Union[str, MarkupContent]]) -> s
 
 
 def format_inlay_hint_label(inlay_hint: InlayHint, session: Session, phantom_uuid: str) -> str:
+    tooltip = format_inlay_hint_tooltip(inlay_hint.get("tooltip"))
     result = ""
     can_resolve_inlay_hint = session.has_capability('inlayHintProvider.resolveProvider')
     label = inlay_hint['label']
     is_clickable = bool(inlay_hint.get('textEdits')) or can_resolve_inlay_hint
     if isinstance(label, str):
         if is_clickable:
-            inlay_hint_click_command = sublime.command_url('lsp_inlay_hint_click', {
-                'session_name': session.config.name,
-                'inlay_hint': cast(dict, inlay_hint),
-                'phantom_uuid': phantom_uuid
+            inlay_hint_click_command = sublime.command_url('lsp_double_click', {
+                'command': 'lsp_inlay_hint_click',
+                'args': {
+                    'session_name': session.config.name,
+                    'inlay_hint': cast(dict, inlay_hint),
+                    'phantom_uuid': phantom_uuid
+                }
             })
             result += '<a href="{command}">'.format(command=inlay_hint_click_command)
-        result += html.escape(label)
+        result += '<span title="{tooltip}">{value}</span>'.format(
+            tooltip=tooltip + '(Double-click to execute)' if is_clickable else tooltip,
+            value=html.escape(label)
+        )
         if is_clickable:
             result += "</a>"
         return result
 
     for label_part in label:
         value = ""
+        tooltip = format_inlay_hint_tooltip(label_part.get("tooltip"))
         is_clickable = is_clickable or bool(label_part.get('command'))
         if is_clickable:
-            inlay_hint_click_command = sublime.command_url('lsp_inlay_hint_click', {
-                'session_name': session.config.name,
-                'inlay_hint': cast(dict, inlay_hint),
-                'phantom_uuid': phantom_uuid,
-                'label_part': cast(dict, label_part)
+            inlay_hint_click_command = sublime.command_url('lsp_double_click', {
+                'command': 'lsp_inlay_hint_click',
+                'args': {
+                    'session_name': session.config.name,
+                    'inlay_hint': cast(dict, inlay_hint),
+                    'phantom_uuid': phantom_uuid,
+                    'label_part': cast(dict, label_part)
+                }
             })
             value += '<a href="{command}">'.format(command=inlay_hint_click_command)
         value += html.escape(label_part['value'])
         if is_clickable:
             value += "</a>"
         # InlayHintLabelPart.location is not supported
         result += "<span title=\"{tooltip}\">{value}</span>".format(
-            tooltip=format_inlay_hint_tooltip(label_part.get("tooltip")),
+            tooltip=tooltip + '(Double-click to execute)' if is_clickable else tooltip,
```

Comment:
There is currently no space between the original tooltip and the addition. So it will look like `Original tooltip(Double-click to execute)`.

I'm also not confident that this addition is really needed. Instead we could add a sentence about it in the LSP docs, that you can insert inlay hints with double click (if supported by the server).

Or alternatively, use "Double click to insert" only if there is no original tooltip from the server.

Reply:
[6e48d6a](https://github.com/sublimelsp/LSP/pull/2175/commits/6e48d6afab925497b72b2590d913d5bd648baf58)

Reply:
> Or alternatively, use "Double click to insert" only if there is no original tooltip from the server.

I'm not sure about this distinction because server is not gonna say how it can be inserted so it can still be a case that is confusing.

But I guess I'll give up here as it's not that important.

Reply:
My idea was that if the server provides a tooltip, then the available space is already taken, because "Server's tooltip text (Double-click to insert)" looks a bit ugly and we don't know what the server uses as the tooltip text (maybe it could even result in more confusing confusing combinations with the server's tooltip text). And with the mouse cursor being the hand/finger icon, users should still easily find out that you can double-click the inlay hint (if supported by the server), so this tooltip hint probably doesn't provide a lot of value.

Reply:
Honestly I don't agree with the statement that "with the mouse cursor being the hand/finger icon, users should still easily find out that you can double-click the inlay hint". At least when talking about myself, I'm not accustomed to having to double click when cursor uses a hand icon. I can not think of any other app/place where that would be the case. Certainly not in browsers with which most people are familiar with.

I have just tried this change with mocked, clickable payload just a while ago and even confused me since I've clicked and I didn't see any reaction so I've started looking through the document to check if anything has changed. For people that didn't see this PR or are not that familiar what inline hints are supposed to do on clicking it will be even more confusing.

Maybe I can suggest something else - no extra text on the tooltip but instead show the message in the status field if the lsp_on_double_click timeouts on single click?

Reply:
Hm, okay I could indeed not find any example of another application where you need to double click when the hand icon is shown. ST's minihtml doesn't support the CSS `cursor` property, though.

I searched a bit on GitHub how inlay hint tooltips are used, and VSCode seems to render a horizontal separator between the tooltip and the double click message:

![tooltip](https://user-images.githubusercontent.com/15612932/180425384-a8fc9ad3-4184-4e34-80db-86305e344e95.png)

We cannot do that, because ST's tooltips don't support HTML markup, but it seems that you can force a linebreak by including `\n` in the "title" string:

![phantom](https://user-images.githubusercontent.com/6579999/215217366-7784b061-971b-478e-8e9a-b3e806d9cc2c.png)

So maybe we could use that, i.e. always show the "Double click to insert" message, but if the server also provides a tooltip, then use `server_tooltip + "\nDouble click to insert"`. Not sure if that would be a better alternative than just a space.

Also servers can use MarkupContent for the tooltip, for example see https://github.com/sjsone/vscode-neos-fusion-lsp/blob/80ea71b2d23c05abbcf47c51711888669b915be4/server/src/languageFeatures/InlayHintLanguageFeature.ts#L43-L51 where it's used for a fenced code block. That would be rendered as literal characters in ST...

Reply:
I would be fine with a newline but also I'm having second thoughts about using title for that. It's quite shouty in ST being all yellow and weird looking. It might seem a little too much for servers that set text edits. And user just needs to be educated about it and not necessarily shown on every occasion where the inlay hint is hovered. In that sense I think my suggestion about showing status text on timeout might make sense.

Reply:
> being all yellow and weird looking

It's controlled by the theme, and has a white background in the Default, Default Dark and Adaptive themes. For the beautiful tooltips from my second screenshots, try out my "Brackets" theme :)

But I'm fine with trying out the status message on timeout. The implementation should be made generic in that case, i.e. I would pass another optional argument `timeout_message: str = ""` for the `lsp_on_double_click` command, so that it is only shown if this message is defined.

Reply:
For now I just pushed a commit to always append the instruction text if a inlay hint contains a command or text edit.
That way the user will know if the inlay hint will insert text, or if a command will be executed as part of double clicking phantoms.

I also tough to completely remove the instruction text from the `title` and just put that info in the release notes and tell users that they have to double click phantoms. (note only one server supports text edits for inlay hints, and not many people use it, don't know any server that supports commands... so I would not put to much effort and thoughts into this).